master/buildbot/util/service.py

```diff
@@ -255,20 +271,85 @@ def getConfigDict(self):
     def reconfigServiceWithBuildbotConfig(self, new_config):
         # get from the config object its sibling config
         config_sibling = getattr(new_config, self.config_attr)[self.name]
+        return self.reconfigServiceWithSibling(config_sibling)
 
+    def reconfigServiceWithSibling(self, sibling):
         # only reconfigure if different as ComparableMixin says.
-        if self.configured and config_sibling == self:
+        if self.configured and sibling == self:
             return defer.succeed(None)
         self.configured = True
-        return self.reconfigService(*config_sibling._config_args,
-                                    **config_sibling._config_kwargs)
+        return self.reconfigService(*sibling._config_args,
+                                    **sibling._config_kwargs)
 
-    def setServiceParent(self, parent):
-        self.master = parent
-        return AsyncService.setServiceParent(self, parent)
+    @defer.inlineCallbacks
+    def startService(self):
+        if not self.configured:
+            # reconfigServiceWithSibling with self, means first configuration
+            yield self.reconfigServiceWithSibling(self)
+        yield AsyncMultiService.startService(self)
 
     def checkConfig(self, *args, **kwargs):
         return defer.succeed(True)
 
     def reconfigService(self, *args, **kwargs):
         return defer.succeed(None)
+
+
+class BuildbotServiceManager(AsyncMultiService, config.ConfiguredMixin,
```

Comment:
BuildbotServiceManager manages BuildbotServices. In the begining, BuildbotService managed itself, but I'd figure out its better to have a higher level manager which manages added, and removed services in reconfigs.

Indeed I should remove the reconfigServiceWithBuildbotConfig in BuildbotService, which will remove config_attrs. compare_attr is a bad copy paste in BuildbotServiceManager